src/main/zapHomeFiles/hud/tools/websockets.js

```diff
@@ -0,0 +1,126 @@
+ */
+
+var WebSockets = (function() {
+
+	// Constants
+	var NAME = "websockets";
+	var LABEL = I18n.t("websockets_tool");
+	var ICONS = {};
+        ICONS.CLOCK = "clock.png";	// Not actually shown anywhere, should be changed if it is going to be shown
+    var tool = {};
+
+	function initializeStorage() {
+		tool.name = NAME;
+        tool.label = LABEL;
+		tool.data = 0;
+		tool.icon = ICONS.CLOCK;
+        tool.isSelected = false;
+        tool.isHidden = true;
+		tool.panel = "";
+        tool.position = 0;
+        tool.messages = [];
+
+        utils.writeTool(tool);
+        registerForZapEvents("org.zaproxy.zap.extension.websocket.WebSocketEventPublisher");
+	}
+
+	function showOptions(tabId) {
+		var config = {};
+
+		config.tool = NAME;
+		config.toolLabel = LABEL;
+		config.options = {remove: I18n.t("common_remove")};
+
+		utils.messageFrame(tabId, "display", {action:"showButtonOptions", config:config})
+			.then(response => {
+				// Handle button choice
+				if (response.id == "remove") {
+					utils.removeToolFromPanel(tabId, NAME);
+				}
+			})
+			.catch(utils.errorHandler);
+	}
+    
+    self.addEventListener("org.zaproxy.zap.extension.websocket.WebSocketEventPublisher", event => {
+		var eventType = event.detail['event.type'];
+        
+        if (eventType == "ws.message") {
+	        let message = {};
+	        
+	        let date = new Date(Number(event.detail.timeSentInMs));
+			let dateStr = date.getHours() + ':' + date.getMinutes() + ':' + date.getSeconds() + '.' + date.getMilliseconds();
+	
+			message.timeInMs = event.detail.timeSentInMs;
+			message.time = dateStr;
+	        message.direction = event.detail.direction;
+	        message.length = event.detail.length;
+	        message.messageSummary = event.detail.messageSummary;
+	        message.opCode = event.detail.opCode + '=' + event.detail.opCodeString;
+			message.id = event.detail.messageId;
+	
+	        utils.messageAllTabs('drawer', {action: 'updateWebSockets', messages: [message]})
+	            .catch(utils.errorHandler);
+	
+			tool.messages.push(message);
+			utils.writeTool(tool);
+	    }
+	});
+
+	self.addEventListener("activate", event => {
+		initializeStorage();
+	});
+
+	function trimMessages(lastPageUnloadTime) {
+		utils.loadTool(NAME)
+			.then(tool => {
+				tool.messages = tool.messages.filter(message => message.timeInMs > lastPageUnloadTime)
+				utils.writeTool(tool)
+			})
+			.catch(utils.errorHandler);
+	}
+
+	self.addEventListener("message", event => {
+		var message = event.data;
+
+		// Broadcasts
+		switch(message.action) {
+			case "initializeTools":
+				initializeStorage();
+				break;
+			
+			case "unload":
+				trimMessages(message.time)
+				break;
+                
+			default:
+				break;
+		}
+
+		// Directed
+		if (message.tool === NAME) {
+			switch(message.action) {
+				case "buttonMenuCicked":
+					showOptions(message.tabId);
+					break;
+
+				case "showWebSocketHttpMessageDetails":
```

Comment:
Should be `showWsMessageDetails`?

Reply:
Will change display.js to as I think the longer version is better in this case